packages/loader/container-loader/src/container.ts

```diff
@@ -627,11 +649,7 @@ export class Container
 	}
 
 	private get connected(): boolean {
-		// "connected" event is delayed until runtime is fully loaded - see setLoaded()
-		return (
-			this.connectionStateHandler.connectionState === ConnectionState.Connected &&
-			this._lifecycleState === "loaded"
-		);
+		return this.connectionStateHandler.connectionState === ConnectionState.Connected;
```

Comment:
FYI (note for reviewers): this is how code looked like before my previous change, which delayed expose of connected state and clientId changes via different mechanism (hiding newly established clientId and the fact that we were connected). This change supersedes it, as it actually delays move to "connected" state at deeper layer (ConnectionStateHandler), and this this is not needed any more.